content/en/blog/_posts/2022-11-21-devicemanager-ga.md/index.md

```diff
@@ -0,0 +1,95 @@
+  1. `PreStartContainer` is called before each container start if indicated by
+     device plugins during registration phase. It allows Device Plugins to run device
+     specific operations on the Devices requested.
+
+Pull Requests that introduced these changes are here:
+1. [Invoke preStart RPC call before container start, if desired by plugin](https://github.com/kubernetes/kubernetes/pull/58282)
+1. [Add GetPreferredAllocation() call to the v1beta1 device plugin API](https://github.com/kubernetes/kubernetes/pull/92665)
+
+With introduction of the above endpoints the interaction between Device Manager in
+Kubelet and Device Manager can be shown as below:
+
+![Process](deviceplugin-framework-overview.png)
+
+### Change in semantics of device plugin registration process 
+Device plugin code was refactored to separate 'plugin' package under the `devicemanager`
+package to lay the groundwork for introducing a `v1beta2` device plugin API. This would
+allow adding support in `devicemanager` to service multiple device plugin APIs at the
+same time.
+
+With this refactoring work, it is now mandatory for a device plugin to start serving its gRPC
+service before registering itself with Kubelet. Previously, these two operations were asynchornous
+and device plugin could register itself before starting its gRPC server which is no longer the
+case. For more details, refer to [PR #109016](https://github.com/kubernetes/kubernetes/pull/109016) and [Issue #112395](https://github.com/kubernetes/kubernetes/issues/112395).
+
+### Dynamic resource allocation
+In Kubernetes 1.26, inspired by how [Persistent Volumes](/docs/concepts/storage/persistent-volumes) are handled in Kubernetes, [Dynamic Resource Allocation](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/3063-dynamic-resource-allocation) has been introduced to cater to devices that have more sophisticated resource requirements
+like:
+1. Decouple device initialization and allocation from the pod lifecycle.
+1. Facilitate dynamic sharing of devices between containers and pods.
+1. Support custom resource-specific parameters
+1. Enable resource-specific setup and cleanup actions
+1. Enable support for Network-attached resources, not just node-local resources
+
+## Is the Device Plugin API stable now?
+No, the Device Plugin API is still not stable, the latest Device Plugin API version 
+available is `v1beta1`. There are plans in the community to introduce `v1beta2` API
+to service multiple plugin APIs at once. A per-API call with request/response types would allow adding support for
+newer API versions without explicitly bumping the API.
+
+In addition to that, there are existing proposals in the community to introduce additional endpoints [KEP-3162: Add Deallocate and PostStopContainer to Device Manager API](https://github.com/kubernetes/kubernetes/pull/109016).
+
+For more details, refer to the slack thread [here](https://kubernetes.slack.com/archives/C0BP8PW9G/p1667481109075849).
```

Comment:
It's not really reasonable to expect readers to be members of Kubernetes' Slack workspace. Instead, summarize the reasons.

If there is an ambition to stabilize the API for Device Plugins, outline that ambition.

Reply:
I have summarized most of the relevant information here already but thought it would nice to put the reference to where the discussion happened. I will remove the link to the slack thread as you rightly pointed out not everyone will be member of the slack workspace.